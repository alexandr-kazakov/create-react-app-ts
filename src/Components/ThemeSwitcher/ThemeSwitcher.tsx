
import { useTheme } from '../../providers/ThemeProvider';

const ThemeSwitcher: React.FC = () => {
  const { theme, toggleTheme } = useTheme();

  return (
    <button onClick={toggleTheme}>
      Toggle Theme 
      <span>({theme})</span>
    </button>
  );
};

export default ThemeSwitcher;
