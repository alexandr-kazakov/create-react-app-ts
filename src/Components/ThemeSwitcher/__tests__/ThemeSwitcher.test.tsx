import { render, fireEvent, screen } from '@testing-library/react';
import ThemeSwitcher from '../ThemeSwitcher';

describe('ThemeSwitcher component', () => {
  test('render', () => {
    render(<ThemeSwitcher />);
    expect(screen.getByText('Toggle Theme')).toBeInTheDocument();
  });

  test('calls onClick when clicked', () => {
    render(<ThemeSwitcher />);
    fireEvent.click(screen.getByText('Toggle Theme'));
    expect(screen.getByText('Toggle Theme')).toBeInTheDocument(); 
  });
});

describe('Button component snapshot', () => {
    test('matches snapshot', () => {
      const { asFragment } = render(<ThemeSwitcher />);
      expect(asFragment()).toMatchSnapshot();
    });
});