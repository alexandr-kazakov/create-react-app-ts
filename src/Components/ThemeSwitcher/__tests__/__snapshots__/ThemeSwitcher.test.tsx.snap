// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot matches snapshot 1`] = `
<DocumentFragment>
  <button>
    Toggle Theme
    <span>
      (light)
    </span>
  </button>
</DocumentFragment>
`;
